walletrpc/service.proto

```diff
@@ -8,18 +8,32 @@ option go_package = "lightwalletd/walletrpc";
 option swift_prefix = "";
 import "compact_formats.proto";
 
+// An identifier for a Zcash value pool.
+enum PoolType {
+    POOL_TYPE_INVALID = 0;
+    TRANSPARENT = 1;
+    SAPLING = 2;
+    ORCHARD = 3;
+}
+
 // A BlockID message contains identifiers to select a block: a height or a
 // hash. Specification by hash is not implemented, but may be in the future.
 message BlockID {
-     uint64 height = 1;
-     bytes hash = 2;
+    uint64 height = 1;
+    bytes hash = 2;
 }
 
 // BlockRange specifies a series of blocks from start to end inclusive.
 // Both BlockIDs must be heights; specification by hash is not yet supported.
+//
+// A server may optionally prune `CompactBlocks` returned to include only data
+// relevant to the requested pool types. If no pool types are specified, the
+// server should default to the legacy behavior of returning only data relevant
+// to the shielded (Sapling and Orchard) pools.
 message BlockRange {
     BlockID start = 1;
     BlockID end = 2;
+    repeated PoolType poolTypes = 3;
```

Comment:
- `BlockRange` is also used in `GetBlockRangeNullifiers`, where setting `PoolType.TRANSPARENT` would be nonsensical. That needs to be forbidden somewhere.
- `BlockRange` is also used in `TransparentAddressBlockFilter`, where this field does not make sense. It should either be required to be unset, or required to be just `PoolType.TRANSPARENT` if set. The documentation for `BlockRange` also needs to be updated to not incorrectly imply that it is only relevant to `CompactBlock`s.